### PR TITLE
Patch balsamic order filtering

### PR DESCRIPTION
**Pull request by islean**
## Description

Currently, Balsamic-qc and balsamic-umi orders do not show up under the balsamic orders tab in cigrid. This PR changes the filtering done on workflow to mirror the analysis filtering in trailblazer.

### Fixed

- Filtering orders on Balsamic only matches on startswith.

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/store/filters/status_order_filters.py [hunk 1]
```diff
@@ -4,11 +4,14 @@
...
 from sqlalchemy.orm import Query
 
+from cg.constants import Workflow
 from cg.server.dto.orders.orders_request import OrderSortField, SortOrder
 from cg.store.models import Customer, Order
 
 
 def filter_orders_by_workflow(orders: Query, workflow: str | None, **kwargs) -> Query:
+    if workflow == Workflow.BALSAMIC:
+        return orders.filter(Order.workflow.startswith(workflow))
```
Comment (islean) [inline cg/store/filters/status_order_filters.py:14]: Should this be fixed here? Or should the call to the apply_order_filter be modified? Will we ever need a stricter balsamic filter which excludes umi and qc?

Comment (seallard) [reply cg/store/filters/status_order_filters.py:14]: No, this is the right spot 👍

**Review by seallard — APPROVED** [review]
👍

**Comment by islean** [comment]
Deployed to cg-services-prod.

**Comment by seallard** [comment]
Closes https://github.com/Clinical-Genomics/cigrid-ui/issues/622

**Pull request merged**